google-ads/src/main/java/com/google/ads/googleads/lib/utils/FieldMasks.java

```diff
@@ -119,6 +123,65 @@ private static void compare(
     }
   }
 
+  /**
+   * Gets the field value of a message from a field mask path.
+   *
+   * @param fieldMaskPath The field mask path.
+   * @param entity The entity to retrieve values from.
+   * @return the field Object.
+   */
+  public static Object getFieldValue(String fieldMaskPath, Message entity) {
+    List<String> fieldMaskParts = new ArrayList<>(Arrays.asList(fieldMaskPath.split("\\.")));
+    Message currentEntity = entity;
+    while (!fieldMaskParts.isEmpty()) {
+      String fieldMaskPart = fieldMaskParts.remove(0);
+      if (Objects.isNull(currentEntity)) {
+        throw new IllegalArgumentException(
+            String.format("Cannot get field value. %s is null.", fieldMaskPart));
+      }
+      Descriptor descriptor = currentEntity.getDescriptorForType();
+      FieldDescriptor childField = descriptor.findFieldByName(fieldMaskPart);
+      Object childValue;
+      try {
+        childValue = currentEntity.getField(childField);
+      } catch (NullPointerException e) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Cannot retrieve field value. A matching field was not found after"
+                    + " navigating %s up to %s.",
+                fieldMaskPath, fieldMaskPart));
+      }
+      if (!childField.isRepeated()
+          && childField.getJavaType() == JavaType.MESSAGE
+          && childValue instanceof Message) {
+        // we can recurse.
+        currentEntity = (Message) childValue;
+      } else if (childField.isRepeated() && childValue instanceof List) {
+        if (fieldMaskParts.size() == 0) {
+          return childValue;
+        } else {
+          throw new IllegalArgumentException(
```

Comment:
Up to you, but you could use `Preconditions.checkArgument` here and avoid the if/else.